hacep/src/main/java/org/kie/hacep/consumer/DroolsConsumerHandler.java

```diff
@@ -44,45 +45,85 @@
     private static final Logger logger = LoggerFactory.getLogger(DroolsConsumerHandler.class);
     private Logger loggerForTest;
     private Producer producer;
-    private DefaultSessionSnapShooter snapshooter;
-    private EnvConfig config;
+    private DefaultSessionSnapShooter sessionSnapShooter;
+    private EnvConfig envConfig;
     private KieSessionContext kieSessionContext;
     private CommandHandler commandHandler;
-    private SnapshotInfos infos;
+    private SnapshotInfos snapshotInfos;
     private boolean shutdown;
 
     public DroolsConsumerHandler(Producer producer, EnvConfig envConfig) {
-        this.config = envConfig;
-        this.snapshooter = new DefaultSessionSnapShooter(config);
-        initializeKieSessionFromSnapshot(config);
+        this.envConfig = envConfig;
+        this.sessionSnapShooter = new DefaultSessionSnapShooter(this.envConfig);
+        initializeKieSessionContext(this.envConfig);
         this.producer = producer;
-        commandHandler = new CommandHandler(kieSessionContext, config, producer, snapshooter);
-        if (config.isUnderTest()) {
+        this.commandHandler = new CommandHandler(this.kieSessionContext, this.envConfig, producer, this.sessionSnapShooter);
+        if (this.envConfig.isUnderTest()) {
             loggerForTest = PrinterUtil.getKafkaLoggerForTest(envConfig);
         }
     }
 
-    private void initializeKieSessionFromSnapshot(EnvConfig config) {
-        if(config.isSkipOnDemanSnapshot()) {// if true we reads the snapshots and wait until the first leaderElectionUpdate
-            this.infos = snapshooter.deserialize();
-            this.kieSessionContext = createSessionHolder(infos);
+    private void initializeKieSessionContext(EnvConfig config) {
+        if(config.isSkipOnDemandSnapshot()) {// if true we reads the snapshots and wait until the first leaderElectionUpdate
```

Comment:
Sorry but I do not understand the meaning of `isSkipOnDemandSnapshot`. Is it the snapshot on demand flag?